### DISCUSSION TITLE
Variable aggregator 0.6.9

### DISCUSSION THREAD

**Discussion by Soabbar** [langgenius/dify#4752] [Help]
Hi team,

Not sure if I am having an issue or a misunderstanding of the node Variable aggregator from 0.6.9.
```
🔗 Variable Aggregator
Improved Variable Assigner supports more flexible variable selection. Additionally, the user experience is enhanced by improving the way node connections are made.

```

I aggregate two variables inside, but I only get one of the variable as output, not both of them. One get erased to replace the other.
Is that a bug?

**Comment by patryk20120** [comment]
A variable assigner is basically made to merge a couple of paths into one. The flow cannot go into two or more paths at the same time as there is no such functionality for now. So the block will return only one value from the executed path.

**Reply by Soabbar** [reply]
Yes I agree with you on how it should work, but in my example i have:

How it should be:
```
Variable1 ---- \ 
 ------- > Variable Aggregator ---> should return V1 + V2 as output
Variable2 ---- / 
```
What I have at the moment:
```
Variable1 ---- \ 
 ------- > Variable Aggregator ---> will return V1  as output, never V2
Variable2 ---- / 
```

i am having block1 ---> block2 ---> Variable Aggregator ---> block 3
Block 3 always have block1 variable. never block 2. 

I think there is a bug there

**Reply by HoyiousXBFamily** [reply]
Yes, same question...

**Reply by secondperspective** [reply]
Yeah same here...
